### PR TITLE
Add ReadTopics

### PR DESCRIPTION
**Pull request by Pryz**
This PR adds a `ReadTopics` function which retrieves all the topics available on the current connection.

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -953,6 +959,68 @@ func (c *Conn) ReadPartitions(topics ...string) (partitions []Partition, err err
...
+				for i, id := range ids {
+					b[i] = brokers[id]
+				}
+				return b
+			}
+
+			topics = make([]Topic, 0, len(res.Topics))
+			for _, t := range res.Topics {
+				if t.TopicErrorCode != 0 && (c.topic == "" || t.TopicName == c.topic) {
+					// We only report errors if they happened for the topic of
```
Comment (stevevls) [inline conn.go]: This error handling seems a little weird to me.  When would we have this case and why would we want to return the empty partitions slice?  It feels like we're losing information and hiding things from the caller.

File: conn_test.go [hunk 2]
```diff
@@ -981,6 +985,38 @@ func testBrokers(t *testing.T, conn *Conn) {
 	}
 }
 
+func testReadTopics(t *testing.T, conn *Conn) {
+	time.Sleep(2 * time.Second)
```
Comment (stevevls) [inline conn_test.go]: What is this sleep playing at?

File: metadata.go [hunk 3]
```diff
@@ -9,7 +9,11 @@ func (r topicMetadataRequestV1) size() int32 {
 }
 
 func (r topicMetadataRequestV1) writeTo(w *bufio.Writer) {
-	writeStringArray(w, []string(r))
+	if len([]string(r)) > 0 {
```
Comment (stevevls) [inline metadata.go]: Heh heh...in some places, we get by without this check when serializing arrays b/c kafka usually don't differentiate between null and empty.

File: conn_test.go (hunk 2 above)
Comment (Pryz) [reply conn_test.go]: gonna remove that. Tests were failing locally because of my local kafka state.

File: metadata.go (hunk 3 above)
Comment (Pryz) [reply metadata.go]: interesting. gonna test without but it was failing for me.

Comment (stevevls) [reply metadata.go]: Oh, I believe that it was failing.  Just saying that Kafka is inconsistent on this front!

**Comment by achille-roussel** [comment]
@Pryz is this still needed?

**Comment by stevevls** [comment]
Should we close this PR?

**Comment by furkanyilmazx** [comment]
I need this feature.

**Comment by Pryz** [comment]
@furkanyilmazx : you can get the same result with https://godoc.org/github.com/segmentio/kafka-go#Conn.ReadPartitions.

**Comment by furkanyilmazx** [comment]
> @furkanyilmazx : you can get the same result with https://godoc.org/github.com/segmentio/kafka-go#Conn.ReadPartitions.
> 
> @stevevls : yes but I will need to fix the ReadPartitions so it works when we don't pass any topic first. Will try to tackle that this week.

I think, 'ReadPartitions' doesn't suitable method to read topics. And for now, either it doesn't work or i couldn't deal with it  :)

**Comment by achille-roussel** [comment]
@furkanyilmazx can you describe the issue you were seeing with `ReadPartitions`? Maybe share a code sample that we can use to track down what's wrong?

**Comment by achille-roussel** [comment]
I think the issue may have been fixed by https://github.com/segmentio/kafka-go/pull/383

I'll close, but feel free to re-open if you're still having issues with `Conn.ReadPartitions`.

**Pull request closed without merging**